elliptic-curve/src/hash2field/expand_msg.rs

```diff
@@ -66,10 +73,18 @@ impl<L: ArrayLength<u8>> Domain<L> {
         }
     }
 
-    pub fn len(&self) -> usize {
+    pub fn len(&self) -> u8 {
         match self {
-            Self::Hashed(_) => L::to_usize(),
-            Self::Array(d) => d.len(),
+            // Can't overflow because it's enforced on a type level.
+            Self::Hashed(_) => L::to_u8(),
+            // Can't overflow because it's checked on creation.
+            Self::Array(d) => d.len() as u8,
```

Comment:
Maybe it would be better to actually codify that comment?

```suggestion
            Self::Array(d) => u8::try_from(d.len()).expect("length overflow"),
```